plugins/openselected.lua

```diff
@@ -31,16 +31,34 @@ config.plugins.openselected = common.merge({
   }
 }, config.plugins.openselected)
 
+local function select_word_under_cursor(doc)
+  local line, col = doc:get_selection() -- will be cursor if no selection
+  local text = doc.lines[line]
+  local pattern = "%s'\""
+
+  if not text or text == "\n" then return nil end
+
+  local start_col = text:sub(1,col):find("[^"..pattern.."]")
```

Comment:
If you want to use `find` for the backwards check, you'll have to continue finding until you go past `col`, then use the previous result.